src/status_im/data_store/messages.cljs

```diff
@@ -82,16 +82,18 @@
 
 (defn- get-unviewed-messages
   [public-key]
-  (into {}
-        (map (fn [[chat-id user-statuses]]
-               [chat-id (into #{} (map :message-id) user-statuses)]))
-        (group-by :chat-id
-                  (-> @core/account-realm
-                      (core/get-by-fields
-                       :user-status
-                       :and {:public-key public-key
-                             :status     "received"})
-                      (core/all-clj :user-status)))))
+  (-> @core/account-realm
+      (core/get-by-fields
+       :user-status
+       :and {:public-key public-key
+             :status     "received"})
+      (.reduce (fn [acc msg _ _]
+                 (let [chat-id (aget msg "chat-id")
+                       message-id (aget msg "message-id")]
+                   (if (contains? acc chat-id)
```

Comment:
Perhaps use `(update acc chat-id (fnil conj #{}) message-id)` and get rid of the explicit conditional.

Reply:
nice suggestion, on it

Reply:
and maybe do `(when (and chat-id message-id) ...)` just in case ;)

Reply:
well, we didn't have it and, idk, from code where these statuses are created, it seems that both are always not nil. Or you have encountered such statuses? 
Otherwise, we can put these nil checks literally everywhere. :)

Reply:
`fnil` thing has been applied, thanks

Reply:
good point :)